lib/instrumentation/kafkajs/index.js

```diff
@@ -18,17 +18,20 @@ module.exports = function initialize(agent, kafkajs, _moduleName, shim) {
   }
 
   shim.setLibrary(shim.KAFKA)
-  instrumentConsumer({ shim, kafkajs, recordMethodMetric })
-  instrumentProducer({ shim, kafkajs, recordMethodMetric })
-}
 
-/**
- * Convenience method for logging the tracking metrics for producer and consumer
- *
- * @param {object} params to function
- * @param {Agent} params.agent instance of agent
- * @param {string} params.name name of function getting instrumented
- */
-function recordMethodMetric({ agent, name }) {
-  agent.metrics.getOrCreateMetric(`${KAFKA.PREFIX}/${name}`).incrementCallCount()
+  shim.wrap(kafkajs, 'Kafka', function nrConstructorWrapper(shim, orig) {
+    return function nrConstructor() {
+      const params = shim.argsToArray.apply(shim, arguments)
+      // eslint-disable-next-line new-cap
+      const instance = new orig(...params)
```

Comment:
if you still want to wrap I'd recommend you use `Reflect.construct(orig, arguments)`. The other option is to do `shim.wrapClass`

```js
  shim.wrapClass(kafkajs, 'Kafka', new ClassWrapSpec({
    post: function nrConstructorWrapper(shim, wrappedClass, name, args) {
      this[kafkaCtx] = {
        brokers: args[0].brokers
      }

      shim.wrap(this, 'producer', instrumentProducer)
      shim.wrap(this, 'consumer', instrumentConsumer)

    }
  }))
```